plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,273 @@
+
+    .. math::
+        p &= \frac{3B-A^2}{3} \; , \; q = \frac{9AB-2A^3-27C}{27} \\
+        A &= \frac{Q + Q^2 \beta + Q \alpha + \alpha \Lambda}{Q^2} \;
+            , \; B = \alpha \frac{1 + 2 Q \beta + \Lambda \beta}{Q^2} \;
+            , \; C = \frac{\alpha^2 \beta}{Q^2} \\
+        \alpha &= \cos^2 \theta \;
+            , \; \beta = \left( \frac{c_s}{v_A}\right)^2 \;
+            , \; \Lambda = \left( \frac{k v_{A}}{\omega_{ci}}\right)^2
+
+    References
+    ----------
+    .. [1] PM bellan, Improved basis set for low frequency plasma waves, 2012,
+       JGR, 117, A12219, doi: `10.1029/2012JA017856
+       <https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2012JA017856>`_.
+
+    .. [2] TE Stringer, Low-frequency waves in an unbounded plasma, 1963, JNE,
+       Part C, doi: `10.1088/0368-3281/5/2/304
+       <https://doi.org/10.1088/0368-3281/5/2/304>`_
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.dispersion import two_fluid_dispersion_solution as tfds
+    >>> k = 0.01 * u.m ** -1
+    >>> theta = 30 * u.deg
+    >>> B = 8.3E-9 * u.T
+    >>> n = 5.e6 * u.m ** -3
+    >>> T_e = 1.6e6 * u.K
+    >>> T_i = 4.e5 * u.K
+    >>> z = 1
+    >>> omega = tfds(B=B, k=k, n=n, T_e=T_e, T_i=T_i, theta=theta, z=z)
+    >>> omega
+    {'fast_mode': <Quantity [[1520.5794506]] rad / s>,
+     'alfven_mode': <Quantity [[1261.75471561]] rad / s>,
+     'acoustic_mode': <Quantity [[0.6881521]] rad / s>}
+
+    >>> k_arr = np.linspace(10**-7, 10**-2, 10000) * u.m ** -1
+    >>> theta = np.linspace(5, 85, 100) * u.deg
+    >>> n = 5.e6 * u.m ** -3
+    >>> B = 8.3E-9 * u.T
+    >>> T_e = 1.6e6 * u.K
+    >>> T_i = 4.e5 * u.K
+    >>> z = 1
+    >>> c = 3.e8 * u.m/u.s
+    >>> c_s = pfp.ion_sound_speed(T_e=T_e, T_i=T_i, n_e=z * n, ion='p+')
+    >>> v_A = pfp.Alfven_speed( B, n, ion='p+')
+    >>> omega_ci = pfp.gyrofrequency(B=B, particle='p+', signed=False, Z=z)
+    >>> omega = tfds(n=n, B=B, T_e=T_e, T_i=T_i, theta=theta, z=z, k=k_arr)
+    >>> omega['fast_mode'][:,40]
+    <Quantity [1.61176312e-02, 1.77335334e-01, 3.38688590e-01, ...,
+               1.52030361e+03, 1.52045553e+03, 1.52060745e+03] rad / s>
+    """
+
+    # Required derived parameters
+    # Compute the ion sound speed using the function from
+    # plasmapy.formulary.parameters
+    c_s = pfp.ion_sound_speed(
+        T_e=T_e, T_i=T_i, n_e=z * n, gamma_e=gamma_e, gamma_i=gamma_i, ion=ion
+    )
+
+    # Compute the ion Alfven speed using the function from
+    # plasmapy.formulary.parameters
+    v_A = pfp.Alfven_speed(B, n, ion=ion)
+
+    # Compute the ion gyro frequency using the function from
+    # plasmapy.formulary.parameters
+    omega_ci = pfp.gyrofrequency(B=B, particle="p+", signed=False, Z=z)
+
+    # Compute the electron plasma frequency using the function from
+    # plasmapy.formulary.parameters
+    omega_pe = pfp.plasma_frequency(n=n, particle="e-", z_mean=z)
```

Comment:
```suggestion
    # Calc needed plasma parameters
    n_e = z_mean * n_i
    c_s = pfp.ion_sound_speed(
        T_e=T_e,
        T_i=T_i,
        ion=ion,
        n_e=n_e,
        gamma_e=gamma_e,
        gamma_i=gamma_i,
        z_mean=z_mean,
    )
    v_A = pfp.Alfven_speed(B, n_i, ion=ion, z_mean=z_mean)
    omega_ci = pfp.gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean)
    omega_pe = pfp.plasma_frequency(n=n_e, particle="e-")
```

This does not alter what the original code was doing, but updates the variable names to match what the new function signature is.   Additionally, I'm reducing the comments to improve readability.  I've personally been someone that over comments, but I've learned over time that comments should be reserved for what is not directly apparent from reading the code itself.  Otherwise, you have the risk of having conflicting statements between how your code reads and what the comments say.